### PR TITLE
RedHat NLP Deliverable 1

### PR DESCRIPTION
**Pull request by Ningxiao-Tang**
## Description of Change

Web scrapping.
Analysis of the log data.

## Team Name and Contributors
team RedHat_NLP:
Hong Xin @HongXin123456, Kyle Bryant @kbryant8, Parker Stone @parkerwstone, Ningxiao Tang, Tianze Shan @TianzeShanBU

## Checklist

Before creating your PR, please ensure the following requirements.

### General

- [x] PR does **not** include irrelevant or hidden files (i.e. `__pycache__`, `.vscode`... etc.)
- [x] Changes are in the correct folder
- [x] There are no merge conflicts
- [x] Commit messages are sensible and descriptive

### Code

- [x] Implementation matches proposed design
- [x] Documentation on how to run the code is created/updated
- [x] Code is legible

### CODE REVIEW

**Comment by Ningxiao-Tang** [comment]
deliverable 1

**Review by gallettilance — APPROVED** [review]

File: RedHat_NLP/Deliverable 1.ipynb [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+        "import numpy as np\n",
+        "import dateutil\n",
+        "from dateutil import parser\n",
+        "import textblob\n",
+        "from textblob import TextBlob\n",
+        "import nltk\n",
+        "from textblob import Word\n",
+        "import matplotlib.pyplot as plt\n",
+        "from nltk.stem.snowball import SnowballStemmer\n",
+        "nltk.download('punkt')\n",
```
Comment (MichaelClifford) [inline RedHat_NLP/Deliverable 1.ipynb]: You might want to break the work up into a couple more cells for increased readability. See [this notebook](https://github.com/aicoe-aiops/ocp-ci-analysis/blob/master/notebooks/data-sources/TestGrid/testgrid_EDA.ipynb) for an example. 

There is no strict rules for how to break up code in notebooks, but a good rule of thumb is like, every logical step where you complete a set of data transformations, or generate some new output, put it in its own cell and add a markdown cell below it annotating what the cell above does and why. Again, see the linked notebook for an example of what I'm talking about : )

File: RedHat_NLP/Deliverable 1.ipynb [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+        "\n",
+        "# upload file\n",
+        "uploaded = files.upload()\n",
```
Comment (MichaelClifford) [inline RedHat_NLP/Deliverable 1.ipynb]: Its unclear to me what this is doing? Why is the first step to upload a file? What is being uploaded? and from where? I would think the first step would be to read data from the google cloud storage endpoint we provided.

I think you need a step here that reads the list of URL's from `https://gcswebci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/canary-release-openshift-origin-installer-e2e-aws-4.5-cnv`. I'm guessing that list is whats in this csv being uploaded? But that step should be done programatically and part of the notebook.

File: RedHat_NLP/Deliverable 1.ipynb [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+        "  sum_neg += tmp.word_counts['error']\n",
+        "  sum_pos = tmp.word_counts['success']\n",
+        "  sum_pos += tmp.word_counts['run']\n",
+        "  failures.append(sum_neg)\n",
+        "  successes.append(sum_pos)\n",
+        "\n",
+        "plt.xlabel('Failures')\n",
+        "plt.ylabel('Successes')\n",
+        "plt.scatter(failures, successes, s=20)\n",
+        "plt.show()\n",
```
Comment (MichaelClifford) [inline RedHat_NLP/Deliverable 1.ipynb]: Once you've collected and cleaned your log messages, you should save them to your environment/repo for future use as a json or CSV file.

File: RedHat_NLP/Deliverable 1.ipynb [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
...hVWqpdIC+J6kR9MtTQHmR8TO9PwFYH51Squ44Za7Hj8fH0ldI7cVdR/W/HqQtBg4CXgYfx4GqbdQqGfviIiTKWwSXy7pncUjo3AYWt0dilavy53cBBwPtAE7geuqW05lSJoJrAU+FhEvFY+r888DUH+hkPvtPieriNiR/u4GvkmhO2DXwOZw+ru7ehVW1HDLXVefj4jYFRF9EdEP3MJrXUQ1ux4kNVEIhC9HxDdSsz8PReotFLLbfUo6gsLOtPVVril3ko6UNGvgOfA+4AkKy35pmuxSYF11Kqy44ZZ7PXBJOurkVODFom6FmjOkf/w8Cp8JKKyHiyU1p1vjLgUeqXR9E02SgFuBJyPi+qJR/jwUi4i6egBnA08DPwM+We16KrTMbwY2psemgeUG5lI42uIZ4PvA0dWuNYdl/yqFrpEeCn3Clw233IAoHJ32M+BxoL3a9ee8Hr6UlrOLwhfggqLpP5nWw1PAWdWuf4LWwTsodA11AZ3pcXY9fh5GeviMZjMzy9Rb95GZmY3AoWBmZhmHgpmZZRwKZmaWcSiYmVnGoWBWRFJf0VVDO9PlEIab9t/T38XFVx81m8qmVbsAs0nmYES0lTNhRPzH8b6JpMaI6Bvv/GZ58ZaC2QgkzZS0QdJj6X4UK4vGHSgx/X+R9I9Fw3dLevfA9JKuk7QReLukP5b0SNoiuVlSY3rcLumJ9H7/oxLLaTbAWwpmg82Q1JmePwdcCJwXES9Jmgc8JGl9jO+szyOBhyPiCkm/DVwFrIiIHkn/BPwRhTPOF0bE7wBImv26l8hsDBwKZoMN6j5KF1D7u3RV2X4Kl06eT+ESy2PVR+FibACnA78L/KRwSR5mULgQ278Ab5b0D8C3ge+NcznMxsWhYDayPwJagN9Nv+i3AtNHmL6Xwd2yxdMeKtqPIGBNRHxi6AtIOhE4A/hT4CLgv46/fLOx8T4Fs5EdBexOgfAe4E2jTL8VaJPUIGkRw9+xbANwgaQ3Qnaf4DelLqqGiFgLfAqoi/sC2+ThLQWzkX0Z+BdJjwMdwE9Hmf5HFPZFbKZwu8fHSk0UEZslfYrC3fAaKFy99HLgIPDPqQ3gsC0Jszz5KqlmZpZx95GZmWUcCmZmlnEomJlZxqFgZmYZh4KZmWUcCmZmlnEomJlZxqFgZmaZ/w80b9KczfD0mAAAAABJRU5ErkJggg==\n",
```
Comment (MichaelClifford) [inline RedHat_NLP/Deliverable 1.ipynb]: You should add a markdown cell below this graph explaining what it tells us about our data.

File: RedHat_NLP/Deliverable 1.ipynb [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+            "text/plain": [
```
Comment (MichaelClifford) [inline RedHat_NLP/Deliverable 1.ipynb]: You should add a markdown cell below this graph explaining what it tells us about our data.

**Comment by MichaelClifford** [comment]
Good first PR folks! :smiley: I added a couple of comments for you all to consider. Take a look and let me know if you have any questions or want to discuss these comments further. Happy to chat about it here in the PR, otherwise well connect again during our meeting next week.

**Comment by Ningxiao-Tang** [comment]
update details of the deliverable 1

**Pull request closed without merging**